sos/report/plugins/openshift.py

```diff
@@ -7,151 +7,369 @@
+            # sysadmins than python-style regex
+            regex = translate(self.get_option('podlogs-filter'))
+        else:
+            regex = None
+
+        for pod in pod_list:
+            if regex and not re.match(regex, pod):
+                continue
+            _log_cmd = "oc logs --namespace=%s %s" % (namespace, pod)
             self.add_cmd_output([
-                "oo-accept-node -v",
-                "oo-admin-ctl-gears list",
-                "ls -laZ %s" % self.gear_base_dir,
-                "ls -la %s" % self.node_proxy_dir
-            ])
+                _log_cmd,
+                _log_cmd + " -p"
+            ], subdir=_log_dir)
 
     def postproc(self):
-        # Redact broker's MongoDB credentials:
-        # MONGO_PASSWORD="PasswordForOpenshiftUser"
-        self.do_file_sub('/etc/openshift/broker.conf',
-                         r"(MONGO_PASSWORD\s*=\s*)(.*)",
-                         r"\1*******")
-
-        # Redact session SHA keys:
-        # SESSION_SECRET=0c31...a7c8
-        self.do_file_sub('/etc/openshift/broker.conf',
-                         r"(SESSION_SECRET\s*=\s*)(.*)",
-                         r"\1*******")
-        self.do_file_sub('/etc/openshift/console.conf',
-                         r"(SESSION_SECRET\s*=\s*)(.*)",
-                         r"\1*******")
-
-        # Redact passwords of the form:
-        # plugin.activemq.pool.1.password = Pa$sW0Rd
-        self.do_file_sub(os.path.join(self.mco_config_dir, "server.cfg"),
-                         r"(.*password\s*=\s*)\S+",
-                         r"\1********")
-        self.do_file_sub(os.path.join(self.mco_config_dir, "client.cfg"),
-                         r"(.*password\s*=\s*)\S+",
-                         r"\1********")
-
-        # Redact DNS plugin credentials
-        # Dynect DNS: DYNECT_PASSWORD=s0ME-p4$_w0RD._
-        plugin_dir = '/etc/openshift/plugins.d/'
-        self.do_file_sub(plugin_dir + 'openshift-origin-dns-dynect.conf',
-                         r"(DYNECT_PASSWORD\s*=\s*)(.*)",
-                         r"\1********")
-        # Fog cloud: FOG_RACKSPACE_API_KEY="apikey"
-        self.do_file_sub(plugin_dir + 'openshift-origin-dns-fog.conf',
-                         r"(FOG_RACKSPACE_API_KEY\s*=\s*)(.*)",
-                         r"\1********")
-        # ISC bind: BIND_KEYVALUE="rndc key"
-        self.do_file_sub(plugin_dir + 'openshift-origin-dns-nsupdate.conf',
-                         r"(BIND_KEYVALUE\s*=\s*)(.*)",
-                         r"\1********")
-        # LDAP authentication:  AuthLDAPBindPassword "IShouldNotBeHere"
-        ldap_paths = '/var/www/openshift/(broker|console)/httpd/conf.d/.*'
-        self.do_path_regex_sub(ldap_paths,
-                               r"(AuthLDAPBindPassword)\s*(.*)",
-                               r"\1********")
+
+        # clear any certificate output
+        self.do_cmd_private_sub('oc')
```

Comment:
This will remove certificates from not only `oc ..` commands but of much more ones (i.e. if a namespace or pod will contain those two letters) - but I expect that redundant work does not matter. If it does, `s/oc/^oc/`?

Reply:
That's a fair point, we'd most likely end up with the same end result, but `^oc` is more explicit so +1. Will update.

Reply:
Actually, this would break the obfuscation since the current method appends `.*` in front of whatever cmd regex is provided. We probably want to review the behavior in depth before changing it just for this plugin.

I'm testing another possibility to attempt to avoid redundant scrubbing.

Reply:
Hm right.. Maybe a smaller improvement in adding a space after the string? i.e.

    self.do_cmd_private_sub('oc ')

Not ideal but better.